dbt/scripts/check_sort_dbt_yaml_files.py

```diff
@@ -0,0 +1,437 @@
+    for parent, children in headings:
+        if not is_sorted(children):
+            print(f"In file: {file_path}")
+            print(f"Under parent heading: {parent}")
+            print("Subheadings ('##') not sorted:")
+            for i, child in enumerate(children):
+                if i > 0 and alphanumeric_key(
+                    normalize_string(children[i - 1])
+                ) > alphanumeric_key(normalize_string(child)):
+                    print(f"---> {child}")
+                else:
+                    print(f"- {child}")
+            print("-" * 40)
+            any_unsorted = True
+
+    return (
+        file_path if not is_sorted(parent_headings) or any_unsorted else None
+    )
+
+
+def check_all_files(directory):
+    """
+    Check all files in a given directory for sorted YAML keys and markdown headings.
+
+    Args:
+        directory (str): The directory path to check.
+
+    Returns:
+        tuple: Results of unsorted files and errors for different checks.
+    """
+    unsorted_columns_files = defaultdict(int)
+    unsorted_data_tests_files = defaultdict(int)
+    error_files = []
+    unsorted_md_files = []
+    unsorted_columns_md_files = []
+    unsorted_shared_columns_md_files = []
+    for root, _, files in os.walk(directory):
+        if "venv" in root:
+            continue
+        for file in files:
+            file_path = os.path.join(root, file)
+            if file.endswith(".yaml") or file.endswith(".yml"):
+                unsorted_columns, errors = check_columns(file_path)
+                for key, value in unsorted_columns.items():
+                    unsorted_columns_files[key] += value
+                unsorted_data_tests, errors = check_data_tests(file_path)
+                for key, value in unsorted_data_tests.items():
+                    unsorted_data_tests_files[key] += value
+                if errors:
+                    error_files.extend(errors)
+            elif file == "docs.md":
+                unsorted_md = check_md_file(file_path)
+                if unsorted_md:
+                    unsorted_md_files.append(unsorted_md)
+            elif file == "columns.md":
+                unsorted_columns_md = check_columns_md_file(file_path)
+                if unsorted_columns_md:
+                    unsorted_columns_md_files.append(unsorted_columns_md)
+            elif file == "shared_columns.md":
+                unsorted_shared_columns_md = check_shared_columns_md_file(
+                    file_path
+                )
+                if unsorted_shared_columns_md:
+                    unsorted_shared_columns_md_files.append(
+                        unsorted_shared_columns_md
+                    )
+
+    return (
+        unsorted_columns_files,
+        unsorted_data_tests_files,
+        error_files,
+        unsorted_md_files,
+        unsorted_columns_md_files,
+        unsorted_shared_columns_md_files,
+    )
+
+
+if __name__ == "__main__":
+    directory = "dbt/"
```

Comment:
**[Suggestion, non-blocking]** Per my suggestion above, I think we should call this script from the context of the `dbt` subdirectory, which would allow us to remove this hardcoded directory and just check all files in the current working directory.